src/Collections/MemoryConfigCollection.php

```diff
@@ -221,8 +221,16 @@ public function update($class, $name, $value)
         return $this;
     }
 
+    /**
+     * @param string $class
+     * @param string $name
+     * @param array $value - non-array values are @deprecated 1.12.0
+     */
     public function merge($class, $name, $value)
     {
+        if (!is_array($value)) {
```

Comment:
I've added this to validate work where config calls to deprecated `->update()` was changed to `->set()` (non-array values) and `->merge()` (array values).  It makes sense add a strong typing to `merge()` in CMS5